src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,211 @@
 package core.basesyntax;
 
+import java.util.Collection;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> oldLast = last;
+        final Node<T> newNode = new Node<>(oldLast, value, null);
+        last = newNode;
+        if (oldLast == null) {
+            first = newNode;
+        } else {
+            oldLast.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            final Node<T> newPrev = node(index).prev;
+            final Node<T> newNode = new Node<>(newPrev, value, node(index));
+            node(index).prev = newNode;
+            if (newPrev == null) {
+                first = newNode;
+            } else {
+                newPrev.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
this method functionality can fit in 4 lines of code if you reuse the existing methods